Sources/TuistKit/Services/Cache/CacheWarmService.swift

```diff
@@ -41,7 +43,50 @@ final class CacheWarmService {
         }
     }
 
+    private func cacheFlavor(named flavorName: String?, from config: Config) throws -> TuistCore.Cache.Flavor {
+        let flavors = config.cache.flavors
+        switch flavorName {
+            case .none:
+                guard let defaultFlavor = flavors.first else {
+                    throw CacheWarmServiceError.missingDefaultFlavor
+                }
+                return defaultFlavor
+
+            case let .some(name):
+                guard let flavor = flavors.first(where: { $0.name == name }) else {
+                    throw CacheWarmServiceError.missingFlavor(name: name, availableFlavors: flavors.map(\.name))
+                }
+                return flavor
+        }
+    }
+
     private var currentPath: AbsolutePath {
         FileHandler.shared.currentPath
     }
 }
+
+enum CacheWarmServiceError: FatalError, Equatable {
```

Comment:
I am opened for the suggestions related to error cases naming and descriptions 😊

Reply:
One suggestion that I'd make is moving this to the top. It's been a non-written convention.